src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1301,5 +1301,42 @@ public override TResult Accept<TResult>(SymbolVisitor<TResult> visitor)
         }
 
         #endregion
+
+        public override bool Equals(Symbol obj, TypeCompareKind compareKind)
+        {
+            MethodSymbol other = obj as MethodSymbol;
+            if (other is null) return false;
+
+            if (!ReferenceEquals(this.OriginalDefinition, other.OriginalDefinition))
+            {
+                return false;
+            }
+
+            // This checks if the methods have the same definition and the type parameters on the containing types have been
+            // substituted in the same way.
+            if (!TypeSymbol.Equals(this.ContainingType, other.ContainingType, compareKind)) return false;
+
+            // If both are declarations, then we don't need to check type arguments
```

Comment:
>declarations [](start = 27, length = 12)

The term "declaration" is somewhat confusing here. Consider using "not constructed" instead. Same for variable names.